clippy_lints/src/types.rs

```diff
@@ -1691,6 +1687,21 @@ impl<'tcx> LateLintPass<'tcx> for Casts {
                 lint_numeric_casts(cx, expr, ex, cast_from, cast_to);
             }
 
+            lint_cast_ptr_alignment(cx, expr, cast_from, cast_to);
+        } else if let ExprKind::MethodCall(method_path, _, args, _) = expr.kind {
+            if method_path.ident.name != sym!(cast) {
```

Comment:
This condition can also be a part of the `if_chain` below

Reply:
Sorry, I'm not sure if I know what you mean.

Does it look like this:

```rust
if let ExprKind::Cast(ref ex, cast_to) = expr.kind {
    ...
    lint_cast_ptr_alignment(cx, expr, cast_from, cast_to);
} else if let ExprKind::MethodCall(method_path, _, args, _) = expr.kind {
    if_chain! {
        if method_path.ident.name == sym!(cast);
        if let Some(generic_args) = method_path.args;
        if let [GenericArg::Type(cast_to)] = generic_args.args;
        // There probably is no obvious reason to do this, just to be consistent with `as` cases.
        if !is_hir_ty_cfg_dependant(cx, cast_to);
        then {
            let (cast_from, cast_to) =
                (cx.typeck_results().expr_ty(&args[0]), cx.typeck_results().expr_ty(expr));
            lint_cast_ptr_alignment(cx, expr, cast_from, cast_to);
        }
    }
```

or this?:

```rust
if let ExprKind::Cast(ref ex, cast_to) = expr.kind {
    ...
    lint_cast_ptr_alignment(cx, expr, cast_from, cast_to);
}

if_chain! {
    if let ExprKind::MethodCall(method_path, _, args, _) = expr.kind;
    if method_path.ident.name == sym!(cast);
    if let Some(generic_args) = method_path.args;
    if let [GenericArg::Type(cast_to)] = generic_args.args;
    // There probably is no obvious reason to do this, just to be consistent with `as` cases.
    if !is_hir_ty_cfg_dependant(cx, cast_to);
    then {
        let (cast_from, cast_to) = (cx.typeck_results().expr_ty(&args[0]), cx.typeck_results().expr_ty(expr));
        lint_cast_ptr_alignment(cx, expr, cast_from, cast_to);
    }
}
```

Both look better than the current one though.

Reply:
I had this in mind:

```rust
if let ExprKind::Cast(ref ex, cast_to) = expr.kind {
    ...
    lint_cast_ptr_alignment(cx, expr, cast_from, cast_to);
} else if let ExprKind::MethodCall(method_path, _, args, _) = expr.kind {
    if_chain! {
        if method_path.ident.name != sym!(cast);
        if let Some(generic_args) = method_path.args;
        if let [GenericArg::Type(cast_to)] = generic_args.args;
        // There probably is no obvious reason to do this, just to be consistent with `as` cases.
        if is_hir_ty_cfg_dependant(cx, cast_to);
        then {
            return;
        }
    }
}
```

Reply:
So, my guess was not so far away. Since there will be only one `if_chain!`, I think my first one is the best. What do you think?

Reply:
Oh yes I must have misunderstood it the first time, that's indeed better =)